agent/xds/server.go

```diff
@@ -126,32 +130,34 @@ type Server struct {
 	activeStreams *activeStreamCounters
 }
 
-// activeStreamCounters simply encapsulates two counters accessed atomically to
-// ensure alignment is correct. This further requires that activeStreamCounters
-// be a pointer field.
-// TODO(eculver): refactor to remove xDSv2 refs
+// activeStreamCounters tracks various stream-related metrics.
+// Requires that activeStreamCounters be a pointer field.
 type activeStreamCounters struct {
-	xDSv3 uint64
-	xDSv2 uint64
+	xDSv3           atomic.Uint64
+	unauthenticated atomic.Uint64
```

Comment:
no need for manual alignment with these types:
>[Int64](https://tip.golang.org/pkg/sync/atomic/#Int64) and [Uint64](https://tip.golang.org/pkg/sync/atomic/#Uint64) are automatically aligned to 64-bit boundaries in structs and allocated data, even on 32-bit systems.

src: https://tip.golang.org/doc/go1.19